### PR TITLE
Process chinese language locale for theme's map

### PR DESCRIPTION
**Pull request by yen-tt**
- add support for chinese language locale in google map and mapbox
- update jest test

J=SLAP-1527
TEST=auto

jest test

### CODE REVIEW

File: static/js/theme-map/Util/helpers.js [hunk 1]
```diff
@@ -1,30 +1,37 @@
...
-    const formattedLocaleStr = localeStr.replace('_', '-');
-    if (provider.getSupportedLocales().includes(formattedLocaleStr)) {
-      return formattedLocaleStr;
+  const { language, modifier, region } = parseLocale(localeStr);
+  if(language.length != 2) {
+    return 'en';
+  }
+  const provider = getMapProvider(mapProvider);
+  let formattedLocaleStr = modifier? `${language}-${modifier}` : language;
+  if(provider === GoogleMaps) {
```
Comment (oshi97) [inline static/js/theme-map/Util/helpers.js:22]: spaces after if

File: static/js/theme-map/Util/helpers.js [hunk 2]
```diff
@@ -1,30 +1,37 @@
...
-    return localeStr;
-  } 
-
-  if (localeStr.length > 2) {
-    const provider = getMapProvider(mapProvider);
-    const formattedLocaleStr = localeStr.replace('_', '-');
-    if (provider.getSupportedLocales().includes(formattedLocaleStr)) {
-      return formattedLocaleStr;
+  const { language, modifier, region } = parseLocale(localeStr);
+  if(language.length != 2) {
```
Comment (oshi97) [inline static/js/theme-map/Util/helpers.js:17]: !==

File: static/js/theme-map/Util/helpers.js [hunk 3]
```diff
@@ -1,30 +1,37 @@
...
+    } else if (modifier === 'Hans') {
+      formattedLocaleStr = `${language}-CN`;
     }
-    return localeStr.substring(0, 2);
   }
-  return 'en';
+  if (provider.getSupportedLocales().includes(formattedLocaleStr)) {
+    return formattedLocaleStr;
+  }
+  return language;
```
Comment (oshi97) [inline static/js/theme-map/Util/helpers.js:34]: just double checking we want to return just `zh` if say `zh-hans-cn` is not supported

Comment (yen-tt) [reply static/js/theme-map/Util/helpers.js:34]: are you referring to the case where `zh-hans` is supported but not `zh-hans-cn`, we should return `zh`?

File: static/js/theme-map/Util/helpers.js [hunk 4]
```diff
@@ -1,30 +1,37 @@
...
-    return localeStr;
-  } 
-
-  if (localeStr.length > 2) {
-    const provider = getMapProvider(mapProvider);
-    const formattedLocaleStr = localeStr.replace('_', '-');
-    if (provider.getSupportedLocales().includes(formattedLocaleStr)) {
-      return formattedLocaleStr;
+  const { language, modifier, region } = parseLocale(localeStr);
+  if (language.length !== 2) {
```
Comment (cea2aj) [inline static/js/theme-map/Util/helpers.js:17]: Is defaulting to 'en' necessary here? If you do remove this if statement, remember to update the function description

File: static/js/theme-map/Util/helpers.js (hunk 3 above)
Comment (oshi97) [reply static/js/theme-map/Util/helpers.js:34]: oh I see formattedLocaleStr never is equal to something like `zh-hans-cn`, would it be possible to move the GoogleMaps specific logic here into the provider itself? Maybe a `transformLocaleForProvider` or something

Comment (cea2aj) [reply static/js/theme-map/Util/helpers.js:34]: In the SDK, `zh-hans-cn` would return `zh-CN` for google maps. I think we should have the same behavior here

Comment (oshi97) [reply static/js/theme-map/Util/helpers.js:34]: yeah I misread the code @cea2aj I assumed formattedLocale would just be the locale canonicalized jambo style

File: static/js/theme-map/Util/helpers.js (hunk 4 above)
Comment (yen-tt) [reply static/js/theme-map/Util/helpers.js:17]: the old code default to en if the locale doesn't follow 2 characters lang code (e.g. if user pass in 'mul' or 'r')

**Review by cea2aj — APPROVED** [review]

**Pull request merged**